### PR TITLE
Feature/soca remove cdate

### PR DESCRIPTION
**Pull request by AndrewEichmann-NOAA**
**Description**

References to CDATE as exported variable replaced with `${PDY}${cyc}` in soca 

Files changed:

jobs/JGDAS_GLOBAL_OCEAN_ANALYSIS_POST
jobs/JGDAS_GLOBAL_OCEAN_ANALYSIS_PREP
jobs/JGDAS_GLOBAL_OCEAN_ANALYSIS_RUN

Addresses GDASApp issue https://github.com/NOAA-EMC/GDASApp/issues/237

This will break GDASApp soca/global-workflow tests until GDASApp PR https://github.com/NOAA-EMC/GDASApp/pull/252 with (had to look this word up) concomitant changes is merged.

**Type of change**

- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

**How Has This Been Tested?**

Tested with soca ctests on Orion 
  
**Checklist**

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] My changes generate no new warnings
- [ ] New and existing tests pass with my changes
- [ ] Any dependent changes have been merged and published

### CODE REVIEW

**Comment by AndrewEichmann-NOAA** [comment]
Would suggest @guillaumevernieres and @CoryMartin-NOAA for review

File: jobs/JGDAS_GLOBAL_OCEAN_ANALYSIS_POST [hunk 1]
```diff
@@ -5,7 +5,7 @@ source "${HOMEgfs}/ush/preamble.sh"
 ##############################################
 # make temp directory
 ##############################################
-export DATA=${DATA:-${DATAROOT}/ocnanal_${CDATE}} # TODO (G): Switch to {cyc} when the downstream code is ready
+export DATA=${DATA:-${DATAROOT}/ocnanal_${PDY}${cyc}}
```
Comment (CoryMartin-NOAA) [inline jobs/JGDAS_GLOBAL_OCEAN_ANALYSIS_POST]: I think @aerorahul wants instead this to be `${RUN}ocnanal_${cyc}` per a conversation he and I had yesterday afternoon

**Review by guillaumevernieres — APPROVED** [review]
Looks good to me.

File: jobs/JGDAS_GLOBAL_OCEAN_ANALYSIS_POST (hunk 1 above)
Comment (aerorahul) [reply jobs/JGDAS_GLOBAL_OCEAN_ANALYSIS_POST]: Yes.  Thanks @CoryMartin-NOAA.
`${PDY}` is not available at this time, so the issue is correct on that, that `${CDATE}` should become `${cyc}`.

I also realized yesterday, that our run directories will get clobbered when `gdas` and `gfs` cycles start.  To distinguish between those, it is best to prepend `ocnanal` with a `${RUN}`.

`${RUN}` is set upstream of the job in the ecflow/rocoto job-card.  EMC internally translates that to `${CDUMP}`, which is also not available at this point in the j-job.

Comment (aerorahul) [inline jobs/JGDAS_GLOBAL_OCEAN_ANALYSIS_POST]:
```suggestion
export DATA=${DATA:-${DATAROOT}/${RUN}ocnanal_${cyc}} 
```

File: jobs/JGDAS_GLOBAL_OCEAN_ANALYSIS_PREP [hunk 2]
```diff
@@ -5,7 +5,7 @@ source "${HOMEgfs}/ush/preamble.sh"
 ##############################################
 # make temp directory
 ##############################################
-export DATA=${DATA:-${DATAROOT}/ocnanal_${CDATE}} # TODO (G): Switch to {cyc} when the downstream code is ready
+export DATA=${DATA:-${DATAROOT}/ocnanal_${PDY}${cyc}}
```
Comment (aerorahul) [inline jobs/JGDAS_GLOBAL_OCEAN_ANALYSIS_PREP]:
```suggestion
export DATA=${DATA:-${DATAROOT}/${RUN}ocnanal_${cyc}} 
```

File: jobs/JGDAS_GLOBAL_OCEAN_ANALYSIS_RUN [hunk 3]
```diff
@@ -6,7 +6,7 @@ source "${HOMEgfs}/ush/preamble.sh"
 ##############################################
 # make temp directory
 ##############################################
-export DATA=${DATA:-${DATAROOT}/ocnanal_${CDATE}} # TODO (G): Switch to {cyc} when the downstream code is ready
+export DATA=${DATA:-${DATAROOT}/ocnanal_${PDY}${cyc}} # TODO (G): Switch to {cyc} when the downstream code is ready
```
Comment (aerorahul) [inline jobs/JGDAS_GLOBAL_OCEAN_ANALYSIS_RUN]:
```suggestion
export DATA=${DATA:-${DATAROOT}/${RUN}ocnanal_${cyc}}
```

**Review by aerorahul — CHANGES_REQUESTED** [review]

**Comment by AndrewEichmann-NOAA** [comment]
Tested with GDASApp ctest in soca

**Review by aerorahul — APPROVED** [review]
looks good.
Thanks for taking the suggestions.

**Comment by AndrewEichmann-NOAA** [comment]
Sorry if you're getting bombarded with review requests, github claims it's undoing them, ???

**Comment by aerorahul** [comment]
> Sorry if you're getting bombarded with review requests, github claims it's undoing them, ???

its all good.

**Review by CoryMartin-NOAA — APPROVED** [review]

**Review by guillaumevernieres — APPROVED** [review]
All good, thanks for doing this @AndrewEichmann-NOAA .

We (I) need to remember to submit PR's in the g-w ahead of doing work in the gdasapp. The overheads of adapting g-w changes in the gdasapp isn't completely trivial, even if light years better than before we had proper testing and CI in place ... note to self ... ish

File: jobs/JGDAS_GLOBAL_OCEAN_ANALYSIS_POST (hunk 1 above)
Comment (WalterKolczynski-NOAA) [reply jobs/JGDAS_GLOBAL_OCEAN_ANALYSIS_POST]: Isn't just using `${cyc}` going to introduce the possibility of collisions with other dates? I know the DA is serial, but if `${KEEPDATA}` is on, the old directory will still be sitting there.

Comment (guillaumevernieres) [reply jobs/JGDAS_GLOBAL_OCEAN_ANALYSIS_POST]: > Isn't just using `${cyc}` going to introduce the possibility of collisions with other dates? I know the DA is serial, but if `${KEEPDATA}` is on, the old directory will still be sitting there.

Good point @WalterKolczynski-NOAA . We'll need to have a `${KEEPDATA}` on switch that renames `${RUN}ocnanal_${cyc}}` to something with `PDY` in its name. We'll tackle this when we decide to actually do something in post!

Comment (WalterKolczynski-NOAA) [reply jobs/JGDAS_GLOBAL_OCEAN_ANALYSIS_POST]: Why not just include PDY to begin with? Unclear why @aerorahul asked for it to be removed (`${RUN}` can be added without removing `${PDY}`).

Comment (AndrewEichmann-NOAA) [reply jobs/JGDAS_GLOBAL_OCEAN_ANALYSIS_POST]: @aerorahul @WalterKolczynski-NOAA @guillaumevernieres Here's a problem with this - in the current soca test, RUN gets assigned in config.base, which comes after the DATA assignment in the jjob. Is this contrary to the assumptions within g-w?

Comment (WalterKolczynski-NOAA) [reply jobs/JGDAS_GLOBAL_OCEAN_ANALYSIS_POST]: `$RUN` should be defined in the pre-job script when needed (`job/rocoto/*` or `ecf/scripts/*`), so it should be added to the rocoto pre-script if it is not already there (don't worry about ecf yet), and the test system should replicate that (if it isn't just using the rocoto pre-script).

**Comment by guillaumevernieres** [comment]
@aerorahul , @WalterKolczynski-NOAA ,  do we have a consensus on ```${cyc}``` vs ```${PDY}${cyc}``` ? If yes and the changes in this PR are OK, could you merge? We can't run the automated `CI` in the `GDASApp` without these changes in `develop`.

**Pull request merged**